src/circuit/circuit_frame.rs

```diff
@@ -2167,26 +2066,77 @@ fn apply_continuation<F: PrimeField, CS: ConstraintSystem<F>>(
         ],
     )?;
 
+    // Continuation::Call0
     /////////////////////////////////////////////////////////////////////////////
-    // Continuation::Call (after hash)                                         //
+    let (body_form, closed_env, the_cont) = {
+        let continuation = AllocatedContPtr::by_index(0, &continuation_components);
+        let (_, arg_t, body_t, closed_env) = Ptr::allocate_maybe_fun(
+            &mut cs.namespace(|| "Call0: allocate fun"),
+            store,
+            result.ptr(store).as_ref(),
+        )?;
+
+        let (body_form, _) = car_cdr(&mut cs.namespace(|| "Call0: body_form"), g, &body_t, store)?;
+
+        let args_is_dummy =
+            arg_t.alloc_equal(&mut cs.namespace(|| "args_is_dummy"), &g.dummy_arg_ptr)?;
+
+        let next_exp = AllocatedPtr::pick(
+            &mut cs.namespace(|| "Call0: pick nexp exp"),
+            &args_is_dummy,
+            &body_form,
+            result,
+        )?;
+
+        let result_is_fun = alloc_equal(
+            cs.namespace(|| "Call0: result_is_fun"),
+            function.tag(),
+            &g.fun_tag,
+        )?;
+
+        let the_cont = AllocatedContPtr::pick(
+            &mut cs.namespace(|| "Call0: pick the continuation"),
+            &result_is_fun,
+            &continuation,
+            &g.error_ptr_cont,
+        )?;
+
+        let the_env = AllocatedPtr::pick(
+            &mut cs.namespace(|| "Call0: pick the environment"),
+            &result_is_fun,
+            &closed_env,
+            env,
+        )?;
+
+        (next_exp, the_env, the_cont)
+    };
+    results.add_clauses_cont(
+        ContTag::Call0,
+        &body_form,
+        &closed_env,
+        &the_cont,
+        &g.false_num,
+    );
+
+    // Continuation::Call, newer_cont is allocated
     /////////////////////////////////////////////////////////////////////////////
     let (next, the_cont) = {
         let next_expr = AllocatedPtr::by_index(1, &continuation_components);
         let result_is_fun = alloc_equal(
-            cs.namespace(|| "result_is_fun default using newer continuation"),
+            cs.namespace(|| "Call: result_is_fun"),
```

Comment:
Same comment as with Call0. I don't think we should do this manual namespacing. It opposes the already heavy abstraction of bellperson namespaces.